pkg/tls/certificatemanagement/certificatebundle.go

```diff
@@ -0,0 +1,125 @@
+// limitations under the License.
+
+package certificatemanagement
+
+import (
+	"bytes"
+	"fmt"
+
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type trustedBundle struct {
+	// certificates is a map of key: hash, value: certificate.
+	certificates map[string]CertificateInterface
+}
+
+// CreateTrustedBundle creates a TrustedBundle, which provides standardized methods for mounting a bundle of certificates to trust.
+func CreateTrustedBundle(certificates ...CertificateInterface) TrustedBundle {
+	bundle := &trustedBundle{
+		certificates: make(map[string]CertificateInterface),
+	}
+	bundle.AddCertificates(certificates...)
+	return bundle
+}
+
+// AddCertificates Adds the certificatess to the bundle.
+func (t *trustedBundle) AddCertificates(certificates ...CertificateInterface) {
+	for _, cert := range certificates {
+		if cert != nil {
+			for cert.GetIssuer() != nil {
+				cert = cert.GetIssuer()
+			}
+			hash := rmeta.AnnotationHash(cert.GetCertificatePEM())
+			if _, found := t.certificates[hash]; !found {
+				t.certificates[hash] = cert
+			}
+		}
+	}
+}
+
+func (t *trustedBundle) MountPath() string {
+	return TrustedCertBundleMountPath
+}
+
+func (t *trustedBundle) HashAnnotations() map[string]string {
+	annotations := make(map[string]string)
+	for hash, cert := range t.certificates {
+		annotations[fmt.Sprintf("hash.operator.tigera.io/%s", cert.GetName())] = hash
+	}
+	return annotations
+}
+
+func (t *trustedBundle) VolumeMount() corev1.VolumeMount {
+	return corev1.VolumeMount{
+		Name:      TrustedCertConfigMapName,
+		MountPath: TrustedCertVolumeMountPath,
+		ReadOnly:  true,
+	}
+}
+
+func (t *trustedBundle) Volume() corev1.Volume {
+	return corev1.Volume{
+		Name: TrustedCertConfigMapName,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: TrustedCertConfigMapName},
+			},
+		},
+	}
+}
+
+func (t *trustedBundle) ConfigMap(namespace string) *corev1.ConfigMap {
+	pemBuf := bytes.Buffer{}
+	for _, cert := range t.certificates {
+		pemBuf.WriteString(fmt.Sprintf("# certificate name: %s\n%s\n\n",
+			cert.GetName(), string(cert.GetCertificatePEM()))) // err is always nil
+	}
+	return &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      TrustedCertConfigMapName,
+			Namespace: namespace,
+		},
+		Data: map[string]string{
+			TrustedCertConfigMapKeyName: pemBuf.String(),
+		},
+	}
+}
+
+// NewCertificate creates a new certificate.
+func NewCertificate(name string, pem []byte, issuer CertificateInterface) CertificateInterface {
+	return &certificate{name: name, pem: pem, issuer: issuer}
+}
+
+type certificate struct {
+	name   string
+	issuer CertificateInterface
+	pem    []byte
+}
+
+func (c *certificate) GetCertificatePEM() []byte {
+	return c.pem
```

Comment:
Please correct me if I'm wrong, but I don't think this is quite correct, don't you need to combine the issuer's PEM as well?

So it should be a something like
```
// default to only this certificates PEM, overwrite this variable with the combination if there's an issuer
pem := c.pem
if c.issuer != nil {
  pem = combinePems(c.issuer.GetCertificatePEM(), c.pem)
}

return pem
```

The call to `c.issuer.GetCertificatePEM()` will run the same function, checking if there's an issuer for this certificates issuer and create combine PEM for this pems issuer it that issuer's issuer.